plugins/nodemon/src/tasks/nodemon.ts

```diff
@@ -0,0 +1,72 @@
+import { ToolKitError } from '@dotcom-tool-kit/error'
+import { hookFork, styles } from '@dotcom-tool-kit/logger'
+import { Task } from '@dotcom-tool-kit/types'
+import { NodemonOptions, NodemonSchema } from '@dotcom-tool-kit/types/lib/schema/nodemon'
+import { VaultEnvVars } from '@dotcom-tool-kit/vault'
+import getPort from 'get-port'
+import nodemon from 'nodemon'
+import { Readable } from 'stream'
+
+export default class Nodemon extends Task<typeof NodemonSchema> {
+  static description = ''
+
+  static defaultOptions: NodemonOptions = {
+    entry: './server/app.js'
+  }
+
+  async run(): Promise<void> {
+    const { entry } = this.options
+    const vault = new VaultEnvVars(this.logger, {
+      environment: 'development'
+    })
+
+    const vaultEnv = await vault.get()
+    const port =
+      Number(process.env.PORT) ||
+      (await getPort({
+        port: [3001, 3002, 3003]
+      }))
+
+    if (!entry) {
+      const error = new ToolKitError(
+        `the ${styles.task('Node')} task requires an ${styles.option('entry')} option`
+      )
+      error.details = `this is the entrypoint for your app, e.g. ${styles.filepath('server/app.js')}`
+      throw error
+    }
+
+    this.logger.verbose('starting the child nodemon process...')
+
+    const env = {
+      ...vaultEnv,
+      PORT: port.toString(),
+      ...process.env
+    }
+    nodemon({ script: entry, env, stdout: false })
+    nodemon.on('readable', () => {
+      // These fields aren't specified in the type declaration for some reason
+      const { stdout, stderr } = (nodemon as unknown) as { stdout: Readable; stderr: Readable }
+      hookFork(this.logger, entry, { stdout, stderr })
+    })
+    const nodemonLogger = this.logger.child({ process: 'nodemon' })
+    nodemon.on('log', (msg) => {
+      function nodemonToWinstonLogLevel(level: string): string {
```

Comment:
**nitpick:** I don't think this will have a particularly negative impact if it's not addressed, but this function is redefined each time `nodemon` logs. It's maybe a bit of a micro-optimisation to move it out of here though so happy if you want to ignore.

Reply:
I have a habit of embedding throwaway functions like this right next to the code that uses it so that you don't have to jump around too much when reading the code. It's the kind of thing that I'd hope V8 will optimise away at runtime though, especially when it's not even using any closure state! Maybe that's too optimistic haha I have no idea about the engine's internals